### PR TITLE
feat(eslint-plugin): remove formatting/layout rules

### LINKED ISSUE #8074: 💀 Rule deletions: formatting (layout) rules
## Overview

#8072 tracks _deprecating_ formatting rules as a non-breaking change. As a followup, this issue tracks _removing_ those formatting rules from typescript-eslint altogether. Those rules now live in [eslint-stylistic](https://eslint.style) and as such don't need to exist in this plugin anymore.

As a breaking change for our next major version, let's:

* Delete the files! 🔪 
* Look into removing the `'layout'` property from our internal rule `meta.type` type union
    * Or at least that this should only impact _internal_ rules. We shouldn't negatively impact `eslint-stylistic` or other consumers of our custom rule APIs

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
BREAKING CHANGE: removes many rules.

## PR Checklist

- [x] Addresses an existing open issue: fixes #8074
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Deletes the rules marked as type `'layout'`, as well as handling for them in the config and website tooling.

I confirmed locally that https://github.com/eslint-stylistic/eslint-stylistic _does_ rely on the `'layout'` constituent for `meta.type` (#8695). But they only ever set `type: 'layout'` and later use [`rule as unknown as`](https://github.com/eslint-stylistic/eslint-stylistic/blob/4d0e78ecc222d7af39c01fdd95d08d5af7491754/packages/eslint-plugin-js/utils/createRule.ts#L7). So this can be worked around in userland.

💖

### CODE REVIEW

File: packages/utils/src/ts-eslint/Rule.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export interface RuleMetaData<
    * - `"suggestion"` means the rule is identifying something that could be done in a better way but no errors will occur if the code isn’t changed.
    * - `"layout"` means the rule cares primarily about whitespace, semicolons, commas, and parentheses, all the parts of the program that determine how the code looks rather than how it executes. These rules work on parts of the code that aren’t specified in the AST.
    */
-  type: 'layout' | 'problem' | 'suggestion';
+  type: 'problem' | 'suggestion';
```
Comment (armano2) [inline packages/utils/src/ts-eslint/Rule.ts:78]: we should either keep type or remove documentation above

**Comment by armano2** [comment]
seems good to me

**Review by armano2 — APPROVED** [review]

**Review by bradzacher — COMMENTED** [review]
Delete it all. Yes. Die.

The only suggestion I would have:
like [camelcase](https://typescript-eslint.io/rules/camelcase/) - I think it would be best to leave tombstones behind to forward people to the new rules.

Why? Because people running on older versions will "click to open docs" in their IDE and it will open our site to https://typescript-eslint.io/rules/block-spacing/. They'll get a 404 page and be completely confused and probably file an issue.

Or they'll be using ESLint v8 and notice `comma-dangle` doesn't work as expected with TS constructs and they'll search our repo for a replacement (as is the standard when looking for extension rules) and find no existing rule -then request it cos they didn't read any other docs.

It'll cost us nothing to keep tombstones around but it'll help orient people and cut off useless support requests.

**Review by bradzacher — APPROVED** [review]

**Comment by armano2** [comment]
> Delete it all. Yes. Die.
> 
> The only suggestion I would have: like [camelcase](https://typescript-eslint.io/rules/camelcase/) - I think it would be best to leave tombstones behind to forward people to the new rules.
> 
> Why? Because people running on older versions will "click to open docs" in their IDE and it will open our site to https://typescript-eslint.io/rules/block-spacing/. They'll get a 404 page and be completely confused and probably file an issue.
> 
> Or they'll be using ESLint v8 and notice `comma-dangle` doesn't work as expected with TS constructs and they'll search our repo for a replacement (as is the standard when looking for extension rules) and find no existing rule -then request it cos they didn't read any other docs.
> 
> It'll cost us nothing to keep tombstones around but it'll help orient people and cut off useless support requests.

what do you think about releasing old version of docs at https://v7.typescript-eslint.io/, that would let ppl who still use old version have access to it

**Comment by bradzacher** [comment]
@armano2 we could do that. The problem there is discoverability - the rules are setup to link directly to `typescript-eslint.io/rules/rule-name` so if someone uses the "open docs" command then they're not gonna see the legacy docs.

I'm personally not too miffed about older versions losing docs - it's going to be a small fraction of users that actively need the old docs.
Each time this has been brought up in the past our collective response has been "it's just not worth the effort".

**Pull request merged**